### ISSUE TITLE
Python PMDA Invalid PMID

### ISSUE THREAD

**Issue by myllynen** [performancecopilot/pcp#558]
As discussed at https://groups.io/g/pcp/topic/25040508 the attached trivial Python PMDA might be revealing some sort of reference counting issue (that's the best guess so far, nobody has pointed out issues with the test code nor there has been deeper analysis of the issue). So far known to affect PMDA BCC (only modules which would not use pmdaCache, zero such modules exist so far), not known to affect PMDA Prometheus although it would seem to use the same approach as this test PMDA code.

```
# pminfo -f foo

foo.example
pmNameIndom: indom=499.0 inst=768355280: Unknown or illegal instance identifier
    inst [768355280] value 768355280
    inst [1 or "1"] value 1
    inst [2 or "2"] value 2
```

[pmdafoo.python.txt](https://github.com/performancecopilot/pcp/files/2345993/pmdafoo.python.txt)

**Comment by andreasgerstmayr** [comment]
should be fixed by eaa5122f8785577eb273d5c1b9d4da0062d1b3c7

**Comment by myllynen** [comment]
Closing based on the above, thanks!
